propagators/autoprop/registry.go

```diff
@@ -0,0 +1,90 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoprop // import "go.opentelemetry.io/contrib/propagators/autoprop"
+
+import (
+	"sync"
+
+	"go.opentelemetry.io/contrib/propagators/aws/xray"
+	"go.opentelemetry.io/contrib/propagators/b3"
+	"go.opentelemetry.io/contrib/propagators/jaeger"
+	"go.opentelemetry.io/contrib/propagators/ot"
+	"go.opentelemetry.io/otel/propagation"
+)
+
+// none is the special "propagator" name that means no propagator shall be
+// configured.
+const none = "none"
+
+// envRegistry is the index of all supported environment variable
+// values and their mapping to a TextMapPropagator.
+var envRegistry = &registry{
+	index: map[string]propagation.TextMapPropagator{
+		// W3C Trace Context.
+		"tracecontext": propagation.TraceContext{},
+		// W3C Baggage.
+		"baggage": propagation.Baggage{},
+		// B3 single-header format.
+		"b3": b3.New(),
+		// B3 multi-header format.
+		"b3multi": b3.New(b3.WithInjectEncoding(b3.B3MultipleHeader)),
+		// Jaeger.
+		"jaeger": jaeger.Jaeger{},
+		// AWS X-Ray.
+		"xray": xray.Propagator{},
+		// OpenTracing Trace.
+		"ottrace": ot.OT{},
+
+		// No-op TextMapPropagator.
+		none: propagation.NewCompositeTextMapPropagator(),
+	},
+}
+
+// registry maintains an index of propagator names to TextMapPropagator
+// implementations that is safe for concurrent use by multiple goroutines
+// without additional locking or coordination.
+type registry struct {
+	mu    sync.Mutex
+	index map[string]propagation.TextMapPropagator
```

Comment:
nit:
the name `index` may a little confusing for map/hashmap/dictionary structure as some languages enable to iterate by index (not only by key). How about renaming it e.g. to `dict`?

Reply:
I'll switch to `names`. `dict` also has conflict with other language meaning.